### PR TITLE
Add major README contents

### PR DESCRIPTION
**Pull request by Connoropolous**
- I noticed that the default Rust Zome is a binary Zome, but I think we want it to be a library

- Is everything I state true? :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,126 @@
...
+## Available Commands
+
+`(u)` means the command is as-yet unimplemented.
+
+| Command   | Use                                                                 |
+|-----------|---------------------------------------------------------------------|
+| agent (u) | Starts a Holochain node as an agent                                 |
+| generate  | Generates a new zome and scaffolds the given capabilities           |
+| init      | Initializes a new Holochain app at the given directory              |
+| package   | Builds the current Holochain app into a `.hcpkg` file               |
```
Comment (lucksus) [inline README.md]: The output is currently called `bundle.json`

File: README.md [hunk 2]
```diff
@@ -1,15 +1,126 @@
...
+| unpack    | Unpacks a Holochain bundle into it's original file system structure |
+| web (u)   | Starts a web server for the current Holochain app                   |
+
+## How To Get Started Building An App
+
+In your terminal, change directories to one where you wish to initialize a new Holochain app.
+Run the following, replacing `your_app_name` with your actual app name:
+```shell
+$ hcdev generate your_app_name
```
Comment (lucksus) [inline README.md]: I believe that would be `init`, not `generate`, right @sphinxc0re?

File: README.md [hunk 3]
```diff
@@ -1,15 +1,126 @@
...
+
+The executable part comes from having embedded Base64 encoded WebAssembly code in the file. *What does that mean?* [WebAssembly](https://webassembly.org/) is a fast and secure low-level language.
+Rather than storing the code in its raw format, Holochain expects the code to be [encoded using Base64](https://en.wikipedia.org/wiki/Base64) , so that its as compact as possible.
+
+If you haven't heard of WebAssembly (WASM for short), that's ok. Important to know is that WASM is intended as a "compilation target" for other languages, not a language to write code in. So instead of writing code in WASM, write code in a language that's familiar to you, and [supports WASM](https://github.com/appcypher/awesome-wasm-langs). When it's time to run your code in Holochain, compile it.
+
+In order to avoid having to handcraft this complex JSON structure, with lots of room for error, the `hcdev package` command streamlines the process of taking your "raw" application folder, and packaging it up into the final `.hcpkg` file.
```
Comment (lucksus) [inline README.md]: Yeah, I like introducing our own filename suffix. Did we use .hcpkg before in the Go version or did you just make that up? Anyway, I like it :)

File: README.md [hunk 4]
```diff
@@ -1,15 +1,126 @@
...
+
+The `hcdev package` tool will automate the process of compiling your Zome code, encoding it, and inserting into the `.hcpkg` file. In order to get these benefits, you just need to make sure that you have the right compilation tools installed on the machine you are using the command line tools from, and that you have the proper configuration files in your Zome folders. 
+
+`hcdev package` works with special files called `.build` files. 
+
+### .build files
+In the process of building a `.hcpkg` file, here is what Holochain does.
+- It iterates Zome by Zome adding them to the JSON
+- For each Zome, it looks for a `code` folder
+- Within that code folder, it looks for a `.build` file
```
Comment (lucksus) [inline README.md]: It actually would replace any folder's contents with a build artifact if that folder contains a `.build` file - not only `code` folders in zomes...

**Comment by philipbeadle** [comment]
Totally love it!

File: README.md (hunk 2 above)
Comment (Connoropolous) [reply README.md]: Yes, you're right! oops

File: README.md (hunk 1 above)
Comment (Connoropolous) [reply README.md]: Yeah... you can also customize the name of the output with a `-o` flag though... 

Anyways, I just borrowed that text from what Julian had in the `hcdev help` command

File: README.md (hunk 3 above)
Comment (Connoropolous) [reply README.md]: I didn't make it up, I saw it in use around so I thought it was settled lol

File: README.md (hunk 4 above)
Comment (Connoropolous) [reply README.md]: Right... it wouldn't have to be named `code` so I could generalize that

Comment (Connoropolous) [reply README.md]: Although it is IMPORTANT that it's called `code` (the folder) I think because it uses the folder name as the key for its value

**Comment by Connoropolous** [comment]
Fixed the tests here

File: README.md [hunk 5]
```diff
@@ -1,15 +1,126 @@
...
+
+A Holochain app can be fully contained in a file known as a `.hcpkg` file.
+It is a JSON file, with a particular structure that Holochain can understand, and execute.
+
+This is an unusual JSON file, it is part configuration, and part executable.
+
+The configuration part comes from the `json` files that are throughout your app. One at the top level for the application (`app.json`) and one for each Zome. Ultimately, these get stitched together into a single tree structure in the `.hcpkg` file.
+
+The executable part comes from having embedded Base64 encoded WebAssembly code in the file. *What does that mean?* [WebAssembly](https://webassembly.org/) is a fast and secure low-level language.
+Rather than storing the code in its raw format, Holochain expects the code to be [encoded using Base64](https://en.wikipedia.org/wiki/Base64) , so that its as compact as possible.
```
Comment (thedavidmeister) [inline README.md]: doesn't base64 make things ~4/3x bigger not smaller?

https://stackoverflow.com/questions/4715415/base64-what-is-the-worst-possible-increase-in-space-usage

i assume the base64 is to make the string as portable/safe/legible as possible vs. raw wasm bytes

File: README.md (hunk 3 above)
Comment (thedavidmeister) [reply README.md]: yes i vaguely remember seeing this somewhere

File: README.md (hunk 5 above)
Comment (Connoropolous) [reply README.md]: Ah, yes, I hadn't realized that. Thanks David

**Comment by Connoropolous** [comment]
@thedavidmeister made the change

File: README.md [hunk 6]
```diff
@@ -1,15 +1,126 @@
...
+
+`(u)` means the command is as-yet unimplemented.
+
+| Command   | Use                                                                 |
+|-----------|---------------------------------------------------------------------|
+| agent (u) | Starts a Holochain node as an agent                                 |
+| generate  | Generates a new zome and scaffolds the given capabilities           |
+| init      | Initializes a new Holochain app at the given directory              |
+| package   | Builds the current Holochain app into a `.hcpkg` file               |
+| unpack    | Unpacks a Holochain bundle into it's original file system structure |
```
Comment (pdaoust) [inline README.md]: `s/it's/its/`

File: README.md [hunk 7]
```diff
@@ -1,15 +1,126 @@
...
+Run the following, replacing `your_app_name` with your actual app name:
+```shell
+$ hcdev generate your_app_name
+$ cd your_app_name
+```
+
+We now have the empty shell of a Holochain app. From here, we will want to generate at least one Zome.
+To do this, run the following, replacing `your_zome_name` with a name related to the functionality you wish to develop. For example: `users`.
+```shell
+$ hcdev generate zomes/your_zome_name rust
```
[the remark this answers is not in the document]
Comment (Connoropolous) [reply README.md]: Nah, since it says to replace the 'your_zome_name' bit

File: README.md [hunk 8]
```diff
@@ -1,15 +1,126 @@
...
+2. a `code` folder, which can be compiled into a single `WASM` file with the code for this Zome
+
+In order for Holochain to run your app, you have to build your code into a single packaged file. Those instructions follow.
+
+## What are .hcpkg files?
+
+A Holochain app can be fully contained in a file known as a `.hcpkg` file.
+It is a JSON file, with a particular structure that Holochain can understand, and execute.
+
+This is an unusual JSON file, it is part configuration, and part executable.
```
[the remark this answers is not in the document]
Comment (Connoropolous) [reply README.md]: I'll replace this

File: README.md (hunk 6 above)
Comment (Connoropolous) [reply README.md]: Will replace

**Comment by Connoropolous** [comment]
@pdaoust with `unpack` you can share a bundled file with someone, and they can unpack it to get the whole dev folder again... except the bundled wasm? I forget about that part

**Comment by Connoropolous** [comment]
thanks for comments all. We just need a resolution from @lucksus and @sphinxc0re on whether this `bundle.json` file is a ".hcpkg" file or not?

**Comment by lucksus** [comment]
Yeah, I think it's fine to call it a .hcpkg file. I just read the section again. It actually does not say that the file is a DNA file. Well, in the general case it really is not, but in the example you are talking about it is.

Anyway, I think this is ready for merge! :)

**Pull request merged**